wallet/src/ui.rs

```diff
@@ -457,19 +457,13 @@ impl TransactionHistoryEntry {
                 CapeTransactionKind::Faucet => "faucet".to_string(),
             },
             hash: {
-                entry
-                    .hash
-                    .map(|hash| {
-                        bincode::serialize(&hash)
+                entry.hash.and_then(|hash| {
+                    bincode::serialize(&hash).ok().and_then(|bytes| {
+                        TaggedBase64::new("HASH", &bytes)
                             .ok()
-                            .map(|bytes| {
```

Comment:
These aren't valid cases for `flat_map()`?

Reply:
Not sure. Clippy said

```
error: called `map(..).flatten()` on an `Option`
   --> wallet/src/ui.rs:461:26
    |
461 |                       .hash
    |  __________________________^
462 | |                     .map(|hash| {
463 | |                         bincode::serialize(&hash)
464 | |                             .ok()
...   |
471 | |                     })
472 | |                     .flatten()
    | |______________________________^
    |
    = note: `-D clippy::map-flatten` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten
help: try using `and_then` instead
    |
461 ~                     .hash.and_then(|hash| {
462 +                         bincode::serialize(&hash)
463 +                             .ok()
464 +                             .map(|bytes| {
465 +                                 TaggedBase64::new("HASH", &bytes)
466 +                                     .ok()
  ...

error: called `map(..).flatten()` on an `Option`
   --> wallet/src/ui.rs:464:34
    |
464 |   ...                   .ok()
    |  ____________________________^
465 | | ...                   .map(|bytes| {
466 | | ...                       TaggedBase64::new("HASH", &bytes)
467 | | ...                           .ok()
468 | | ...                           .map(|tb| tb.to_string())
469 | | ...                   })
470 | | ...                   .flatten()
    | |________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten
help: try using `and_then` instead
    |
464 ~                             .ok().and_then(|bytes| {
465 +                                 TaggedBase64::new("HASH", &bytes)
466 +                                     .ok()
467 +                                     .map(|tb| tb.to_string())
468 +                             })
    |

error: could not compile `cape_wallet` due to 2 previous errors
```

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#map_flatten

Reply:
Oh. Huh, yeah, I see.